src/main/java/ch/heigvd/pro/b04/polls/PollController.java

```diff
@@ -1,27 +1,55 @@
 package ch.heigvd.pro.b04.polls;
 
+import ch.heigvd.pro.b04.auth.exceptions.UnknownUserCredentialsException;
 import ch.heigvd.pro.b04.moderators.Moderator;
+import ch.heigvd.pro.b04.moderators.ModeratorRepository;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Optional;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 
 @RestController
 public class PollController {
 
-  @Autowired
-  private final PollRepository repository;
+  private final PollRepository polls;
+  private final ModeratorRepository moderators;
 
-  public PollController(PollRepository repository) {
-    this.repository = repository;
+  public PollController(
+      ModeratorRepository moderators,
+      PollRepository polls
+  ) {
+    this.moderators = moderators;
+    this.polls = polls;
   }
 
-  @RequestMapping(value = "/poll", method = RequestMethod.GET)
-  public List<Poll> all() {
-    return repository.findAll();
+  /**
+   * Returns a {@link List} of all the {@link Poll} instances that are associated with a certain
+   * moderator, with a certain token.
+   *
+   * @param token       The authentication token to use for the moderator.
+   * @param idModerator The identifier of the moderator for which we query the polls.
+   * @return The {@link List} of all the {@link Poll}s of this moderator.
+   * @throws UnknownUserCredentialsException If the moderator is not known, or the credentials are
+   *                                         not valid.
+   */
+  @RequestMapping(value = "/mod/{idModerator}/poll", method = RequestMethod.GET)
```

Comment:
Should be changed in API documentation before implementation, but fine.

Reply:
Referenced in issue #47.